t/t3903-stash.sh

```diff
@@ -287,6 +287,14 @@ test_expect_success 'stash an added file' '
 	test new = "$(cat file3)"
 '
 
+test_expect_success 'stash --intent-to-add file' '
+	git reset --hard &&
+	echo new >file4 &&
+	git add --intent-to-add file4 &&
+	test_must_fail git stash &&
+	git rm -f file4
```

Comment:
I think this should be moved up 1 line and should read `test_when_finished "git rm -f file4"` instead.